plugin/src/main/kotlin/org/gradle/github/dependency/extractor/internal/GitHubRepositorySnapshotBuilder.kt

```diff
@@ -85,17 +85,32 @@ class GitHubRepositorySnapshotBuilder(
     private class DependencyCollector {
         val resolved: MutableMap<String, GitHubDependency> = mutableMapOf()
 
+        /**
+         * Merge each resolved component with the same ID into a single GitHubDependency.
+         */
         fun addResolved(component: ResolvedComponent, rootComponent: ResolvedComponent) {
```

Comment:
Will this method ever be called concurrently from multiple threads?

Reply:
Not as it's currently implemented. We [collect all `ResolvedConfiguration` instances and then iterate in a final step](https://github.com/gradle/github-dependency-extractor/blob/db85ed1befa34f200ed5d44ff942a24b7120c368/plugin/src/main/kotlin/org/gradle/github/dependency/extractor/internal/GitHubRepositorySnapshotBuilder.kt#L51-L56).